### PR TITLE
[PE-2210] add oracle connection support

### PR DESCRIPTION
**Pull request by kevinkirkup**
**Jira Issue**: [PE-2210]

```release-note:
Add Oracle Fast Connect Data Source
```

## Details

[PE-2210]: https://pureport.atlassian.net/browse/PE-2210

### CODE REVIEW

**Comment by mtraynham** [comment]
Bit confused on what's all being merged here.  Why does this include previous commits for develop?

**Comment by kevinkirkup** [comment]
> Bit confused on what's all being merged here. Why does this include previous commits for develop?

There were environment changes need by this branch. If you look at the diff, they shouldn't be included.

**Comment by mtraynham** [comment]
Environment changes?  I'm looking at the diff, but its 17,000 additions and 3,000 deletions and the first few changes are things like the changelog file generation.  Not sure I understand how that is scoped to this PR.

**Comment by kevinkirkup** [comment]
> Environment changes? I'm looking at the diff, but its 17,000 additions and 3,000 deletions and the first few changes are things like the changelog file generation. Not sure I understand how that is scoped to this PR.

Rebased to develop. Try now...

**Comment by kevinkirkup** [comment]
@mtraynham Anything else?

File: website/docs/d/oracle_connection.html.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+The following arguments are supported:
+
+* `connection_id` - (Required) The ID of the connection. You should use the `pureport_connections` data source
+  for querying the list of available connections and discover the ID for the connection.
+
+- - -
+
+* `filter` - (Optional) A filter used to scope the list e.g. by tags.
+  * `name` - (Required) The name of the filter. The valid values are defined in the
+    [Pureport SDK Model](https://github.com/pureport/pureport-sdk-go/blob/develop/docs/client/AwsDirectConnectConnection.md).
```
Comment (joshpoth) [inline website/docs/d/oracle_connection.html.md]: Minor, I wonder if this should be `master` instead
```suggestion
    [Pureport SDK Model](https://github.com/pureport/pureport-sdk-go/blob/master/docs/client/AwsDirectConnectConnection.md).
```

**Comment by kevinkirkup** [comment]
Need to update the resource documentations as well.

**Review by joshpoth — DISMISSED** [review]
Pretty yuuuge but seems OK to me

**Comment by kevinkirkup** [comment]
Updated.

File: website/docs/d/oracle_connection.html.md [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+  * `name` - (Required) The name of the filter. The valid values are defined in the
+    [Pureport SDK Model](https://github.com/pureport/pureport-sdk-go/blob/master/docs/client/AwsDirectConnectConnection.md).
+    Nested values are supported. E.g.("Location.DisplayName")
+  * `values` - (Required) The value of the filter. Currently only regex strings are supported.
+
+## Attributes
+
+* `name` - The name for the connection
+* `location_href` - HREF for the Pureport Location to attach the connection.
+* `network_href` - HREF for the network to associate the connection.
```
Comment (joshpoth) [inline website/docs/d/oracle_connection.html.md:43]: maybe nit picking but this description seems awkward to me.
`HREF for the network the connection belongs to` maybe? or something similar?

**Review by joshpoth — DISMISSED** [review]

File: website/docs/d/oracle_connection.html.md (hunk 2 above)
Comment (kevinkirkup) [reply website/docs/d/oracle_connection.html.md:43]: Updated.

**Review by joshpoth — APPROVED** [review]

**Pull request merged**